test/sql/types/enum/test_alter_enum.test

```diff
@@ -0,0 +1,68 @@
+# name: test/sql/types/enum/test_alter_enum.test
+# description: Alter column type from String to ENUM
+# group: [enum]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE person (
+    name text
+);
+
+statement ok
+insert into person values ('Pedro'), ('Mark'), ('Hannes'), ('Pedro'), ('Pedro'), ('Mark')
+
+statement ok
+CREATE TYPE name_enum AS ENUM ('Pedro', 'Mark');
+
+#This should error out since we can't do this cast
+statement error
+ALTER TABLE person ALTER name TYPE name_enum
+
+statement ok
+DROP TYPE name_enum
+
+statement ok
+CREATE TYPE name_enum AS ENUM ('Pedro', 'Mark', 'Hannes');
+
+statement ok
+ALTER TABLE person ALTER name TYPE name_enum
```

Comment:
Could you add a test to verify that altering to an enum correctly adds a dependency to that enum, so we cannot drop the enum while the table exists?